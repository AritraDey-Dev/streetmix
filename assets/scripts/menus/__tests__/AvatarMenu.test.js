/* eslint-env jest */
import React from 'react'
import { fireEvent } from '@testing-library/react'
import { renderWithRedux } from '../../../../test/helpers/render'
import AvatarMenu from '../AvatarMenu'

const user = {
  id: 'foo'
}

describe('AvatarMenu', () => {
  it('renders user avatar', () => {
    const wrapper = renderWithRedux(<AvatarMenu user={user} />)
    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('renders user avatar for admin', () => {
    const user = {
      id: 'foo',
      roles: ['ADMIN']
    }
    const wrapper = renderWithRedux(<AvatarMenu user={user} />)
    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('calls click handler', () => {
    const onClick = jest.fn()
    const wrapper = renderWithRedux(
      <AvatarMenu user={user} onClick={onClick} />
    )
    fireEvent.click(wrapper.getByText(user.id))
    expect(onClick).toHaveBeenCalled()
  })
})
