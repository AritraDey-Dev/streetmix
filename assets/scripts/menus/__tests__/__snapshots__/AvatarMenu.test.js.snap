// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarMenu renders user avatar 1`] = `
<DocumentFragment>
  <button
    class="menu-attached menu-avatar"
  >
    <div
      class="avatar"
    >
      <img
        alt="foo"
      />
    </div>
    <span
      class="user-id"
    >
      foo
    </span>
  </button>
</DocumentFragment>
`;

exports[`AvatarMenu renders user avatar for admin 1`] = `
<DocumentFragment>
  <button
    class="menu-attached menu-avatar"
  >
    <div
      class="avatar"
    >
      <img
        alt="foo"
      />
    </div>
    <svg
      class="svg-inline--fa menu-avatar-admin fa-crown"
    />
    <span
      class="user-id"
    >
      foo
    </span>
  </button>
</DocumentFragment>
`;
